### PR TITLE
Restore HexPrefix in null cases

### PR DESCRIPTION
**Pull request by Nana-EC**
Signed-off-by: Nana-EC <nana.essilfie-conduah@hedera.com>

**Description**:
In 0.54.0 we updated the `addHexPrefix` flow to return null where applicable.
This does not conform with the expected encoding in JSON RPC cases.
Thus to make return consistent across modules

- Update `addHexPrefix` to return `0x` instead of null
- Updated `toHexString` to return  `0x` instead of null for both nil and empty cases
- Change `indexOf` in `addHexPrefix` to use `substring` based on bench mark tests previously noted
- Update `encodeKey()` to preserve ability to differentiate between null and empty keys
- Expand js test coverage
- Update spec tests

**Related issue(s)**:

Partially addresses #3535 

**Notes for reviewer**:
Ran acceptance tests against integration with local REST changes

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by steven-sheehy — CHANGES_REQUESTED** [review]
We should also fix `ContractFeature.verifyContractExecutionResults()` to not throw a NPE if `contractResult.getFunctionParameters()` is null.

File: hedera-mirror-rest/utils.js [hunk 1]
```diff
@@ -749,7 +749,7 @@ const addHexPrefix = (hexData) => {
  */
 const toHexString = (byteArray, addPrefix = false, padLength = undefined) => {
   if (_.isNil(byteArray)) {
-    return null;
+    return hexPrefix;
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-rest/utils.js:752]: If we're trying to be in accordance with [Hex Value Encoding](https://eth.wiki/json-rpc/API#hex-value-encoding) for byte arrays then the `addPrefix=false` indeed isn't respected
An empty byte array should always return `0x`.
What we're adding is that nil byte arrays should also return `0x`

[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-rest/utils.js:753]: Added logic to encodeKey to preserve logic and adding comments to highlight standardization

File: hedera-mirror-rest/utils.js [hunk 2]
```diff
@@ -733,23 +733,24 @@ const randomString = async (length) => {
...
+ * Logic conforms with ETH hex value encoding, therefore nill and empty return '0x'
  * @param {Array} byteArray Array of bytes to be converted to hex string
  * @param {boolean} addPrefix Whether to add the '0x' prefix to the hex string
  * @param {Number} padLength The length to left pad the result hex string
  * @return {String} Converted hex string
  */
 const toHexString = (byteArray, addPrefix = false, padLength = undefined) => {
-  if (_.isNil(byteArray)) {
-    return null;
+  if (_.isNil(byteArray) || _.isEmpty(byteArray)) {
```
Comment (steven-sheehy) [inline hedera-mirror-rest/utils.js:752]: `isEmpty` also checks if nil.

```suggestion
  if (_.isEmpty(byteArray)) {
```

Comment (Nana-EC) [reply hedera-mirror-rest/utils.js:752]: Yeah recalled after pushing, coming in next update along with functionparams catch

Comment (Nana-EC) [reply hedera-mirror-rest/utils.js]: Updated

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**